opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ListTaskPresenter.java

```diff
@@ -243,33 +247,51 @@ private void onFeatureSelected(Feature feature) {
-                listTaskInteractor.fetchInterventionDetails(code, feature.id(), false);
-            } else if (PAOT.equals(code)) {
-                listTaskInteractor.fetchInterventionDetails(code, feature.id(), false);
-            } else if (org.smartregister.reveal.util.Utils.isFocusInvestigationOrMDA()) {
-                listTaskInteractor.fetchFamilyDetails(selectedFeature.id());
-            } else if (IRS_VERIFICATION.equals(code) && COMPLETE.equals(businessStatus)) {
-                listTaskInteractor.fetchInterventionDetails(IRS_VERIFICATION, feature.id(), false);
+            onFeatureSelectedByNormalClick(feature);
+        }
+    }
+
+    private void onFeatureSelectedByNormalClick(Feature feature) {
+        String businessStatus = getPropertyValue(feature, FEATURE_SELECT_TASK_BUSINESS_STATUS);
+        String code = getPropertyValue(feature, TASK_CODE);
+        selectedFeatureInterventionType = code;
+        if ((IRS.equals(code) || MOSQUITO_COLLECTION.equals(code) || LARVAL_DIPPING.equals(code) || PAOT.equals(code) || REGISTER_FAMILY.equals(code) || IRS_VERIFICATION.equals(code))
+                && (NOT_VISITED.equals(businessStatus) || businessStatus == null)) {
+            if (validateFarStructures()) {
+                validateUserLocation();
+            } else {
+                onLocationValidated();
             }
+        } else if (IRS.equals(code) &&
+                (NOT_SPRAYED.equals(businessStatus) || SPRAYED.equals(businessStatus) || NOT_SPRAYABLE.equals(businessStatus) || PARTIALLY_SPRAYED.equals(businessStatus)
+                        || COMPLETE.equals(businessStatus) || NOT_ELIGIBLE.equals(businessStatus) || NOT_VISITED.equals(businessStatus))) {
+
+            listTaskInteractor.fetchInterventionDetails(IRS, feature.id(), false);
+        } else if ((MOSQUITO_COLLECTION.equals(code) || LARVAL_DIPPING.equals(code))
+                && (INCOMPLETE.equals(businessStatus) || IN_PROGRESS.equals(businessStatus)
+                || NOT_ELIGIBLE.equals(businessStatus) || COMPLETE.equals(businessStatus))) {
+            listTaskInteractor.fetchInterventionDetails(code, feature.id(), false);
+        } else if (PAOT.equals(code)) {
+            listTaskInteractor.fetchInterventionDetails(code, feature.id(), false);
+        } else if (org.smartregister.reveal.util.Utils.isFocusInvestigationOrMDA()) {
+            listTaskInteractor.fetchFamilyDetails(selectedFeature.id());
+        } else if (IRS_VERIFICATION.equals(code) && COMPLETE.equals(businessStatus)) {
+            listTaskInteractor.fetchInterventionDetails(IRS_VERIFICATION, feature.id(), false);
+        }
+    }
+
+    private void onFeatureSelectedByLongClick(Feature feature) {
+        String taskStatus = getPropertyValue(feature, TASK_STATUS);
+        String code = getPropertyValue(feature, TASK_CODE);
+
+        selectedFeatureInterventionType = code;
+        if (READY.toString().equals(taskStatus)) {
```

Comment:
This does not work for Families, its proceeding with setting on Inactive even if there are tasks some of which are members